packages/rum-core/src/domain/foregroundContexts.spec.ts

```diff
@@ -0,0 +1,240 @@
+
+          expect(foregroundContext.getInForeground(relativeNow())).toEqual(false)
+        })
+      })
+
+      describe('getInForegroundPeriods', () => {
+        it('should an empty array', () => {
+          const { clock } = setupBuilder.build()
+
+          clock.tick(1_000)
+
+          expect(foregroundContext.getInForegroundPeriods(relativeNow(), 0 as Duration)).toEqual([])
+        })
+      })
+    })
+
+    describe('with two closed focus period & one active one', () => {
+      /* 
+      events         F      B   F       B   F
+      periods        <------>   <------->   <---- - - -
+      time       0   5  10  15  20  25  30  35  40  45
+      */
+      beforeEach(() => {
+        const { clock } = setupBuilder.build()
+        clock.tick(BLUR_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(FOCUS_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('blur'))
+        clock.tick(BLUR_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(FOCUS_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('blur'))
+        clock.tick(BLUR_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(FOCUS_PERIOD_LENGTH)
+      })
+
+      describe('getInForeground', () => {
+        it('should return false in the first blurred period', () => {
+          expect(foregroundContext.getInForeground(2 as RelativeTime)).toEqual(false)
+        })
+
+        it('should return true in the first focused period', () => {
+          expect(foregroundContext.getInForeground(10 as RelativeTime)).toEqual(true)
+        })
+
+        it('should return false in the second blurred period', () => {
+          expect(foregroundContext.getInForeground(17 as RelativeTime)).toEqual(false)
+        })
+
+        it('should return true in the second focused period', () => {
+          expect(foregroundContext.getInForeground(25 as RelativeTime)).toEqual(true)
+        })
+
+        it('should return false in the third blurred period', () => {
+          expect(foregroundContext.getInForeground(32 as RelativeTime)).toEqual(false)
+        })
+
+        it('after all the events, should return true because the last period was never closed', () => {
+          expect(foregroundContext.getInForeground(42 as RelativeTime)).toEqual(true)
+        })
+      })
+
+      describe('getInForegroundPeriods', () => {
+        describe('for the whole period', () => {
+          let periods: InForegroundPeriod[] | undefined
+          beforeEach(() => {
+            periods = foregroundContext.getInForegroundPeriods(0 as RelativeTime, 50 as Duration)
+          })
+          it('should have 3 in foreground periods', () => {
+            expect(periods).toHaveSize(3)
+          })
+
+          it('should have the whole first period', () => {
+            expect(periods![0]).toEqual({
+              start: toServerDuration(5 as Duration),
```

Comment:
let's not mirror the implementation in the test, `5 * 1e6 as ServerDuration` is what we expect
if we don't want to deal with that here we could also move the conversion to ServerDuration inside `viewCollection`.